src/jit/x64/tiles.dasc

```diff
@@ -263,8 +263,68 @@ MVM_JIT_TILE_DECL(cast) {
 }
 
 
+/* Logic is the same as cast above except loading from memory instead of reg */
+/* See comments in cast above for more detail */
 MVM_JIT_TILE_DECL(cast_load_addr) {
-    DIE("NYI");
+    MVMint32 ofs       = tile->args[0];
+    MVMint32 load_size = tile->args[1];
+    MVMint32 to_size   = tile->args[2];
+    MVMint32 from_size = tile->args[3];
+    MVMint32 cast_mode = tile->args[4];
```

Comment:
Ouch. The size of `args` is limited to 4. I guess that's not enough then.

Reply:
... although you don't strictly need both from_size and load_size, so maybe there is a fix for this

Reply:
If we set the args array size to something higher (6 might be a good number?), would that interfere with any code expecting a certain size or alignment of the structure?

Reply:
Yes, that is actually the easiest and cheapest fix.